.github/workflows/test.yml

```
@@ -0,0 +1,65 @@
+
+jobs:
+  test:
+    name: Build and test
+    runs-on: ubuntu-latest
+    env:
+      # Add environment variables for testing
+      DISCORD_BOT_TOKEN: "test_discord_token_for_ci"
+      MIX_ENV: "test"
+
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)